2019-07-07 13:12:55,189 ERROR: Request finalizing failed with an error while handling an error [in /home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/app.py:1838]
Traceback (most recent call last):
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/app.py", line 1816, in full_dispatch_request
    return self.finalize_request(rv)
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/app.py", line 1833, in finalize_request
    response = self.process_response(response)
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/app.py", line 2114, in process_response
    self.session_interface.save_session(self, ctx.session, response)
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/sessions.py", line 375, in save_session
    val = self.get_signing_serializer(app).dumps(dict(session))
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/itsdangerous.py", line 565, in dumps
    payload = want_bytes(self.dump_payload(obj))
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/itsdangerous.py", line 847, in dump_payload
    json = super(URLSafeSerializerMixin, self).dump_payload(obj)
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/itsdangerous.py", line 550, in dump_payload
    return want_bytes(self.serializer.dumps(obj))
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/json/tag.py", line 296, in dumps
    return dumps(self.tag(value), separators=(',', ':'))
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/json/__init__.py", line 179, in dumps
    rv = _json.dumps(obj, **kwargs)
  File "/usr/lib/python3.7/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/usr/lib/python3.7/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib/python3.7/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/json/__init__.py", line 81, in default
    return _json.JSONEncoder.default(self, o)
  File "/usr/lib/python3.7/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type Project is not JSON serializable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/app.py", line 1833, in finalize_request
    response = self.process_response(response)
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/app.py", line 2114, in process_response
    self.session_interface.save_session(self, ctx.session, response)
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/sessions.py", line 375, in save_session
    val = self.get_signing_serializer(app).dumps(dict(session))
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/itsdangerous.py", line 565, in dumps
    payload = want_bytes(self.dump_payload(obj))
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/itsdangerous.py", line 847, in dump_payload
    json = super(URLSafeSerializerMixin, self).dump_payload(obj)
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/itsdangerous.py", line 550, in dump_payload
    return want_bytes(self.serializer.dumps(obj))
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/json/tag.py", line 296, in dumps
    return dumps(self.tag(value), separators=(',', ':'))
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/json/__init__.py", line 179, in dumps
    rv = _json.dumps(obj, **kwargs)
  File "/usr/lib/python3.7/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/usr/lib/python3.7/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib/python3.7/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/json/__init__.py", line 81, in default
    return _json.JSONEncoder.default(self, o)
  File "/usr/lib/python3.7/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type Project is not JSON serializable
2019-07-07 13:13:09,784 ERROR: Exception on /proj/2 [GET] [in /home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/app.py:1761]
Traceback (most recent call last):
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/app.py", line 1816, in full_dispatch_request
    return self.finalize_request(rv)
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/app.py", line 1833, in finalize_request
    response = self.process_response(response)
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/app.py", line 2114, in process_response
    self.session_interface.save_session(self, ctx.session, response)
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/sessions.py", line 375, in save_session
    val = self.get_signing_serializer(app).dumps(dict(session))
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/itsdangerous.py", line 565, in dumps
    payload = want_bytes(self.dump_payload(obj))
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/itsdangerous.py", line 847, in dump_payload
    json = super(URLSafeSerializerMixin, self).dump_payload(obj)
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/itsdangerous.py", line 550, in dump_payload
    return want_bytes(self.serializer.dumps(obj))
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/json/tag.py", line 296, in dumps
    return dumps(self.tag(value), separators=(',', ':'))
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/json/__init__.py", line 179, in dumps
    rv = _json.dumps(obj, **kwargs)
  File "/usr/lib/python3.7/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/usr/lib/python3.7/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib/python3.7/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/home/greg/Programming/RFI_app/venv/lib/python3.7/site-packages/flask/json/__init__.py", line 81, in default
    return _json.JSONEncoder.default(self, o)
  File "/usr/lib/python3.7/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type Project is not JSON serializable
